### PR TITLE
api: Add Keybind interface

### PR DESCRIPTION
**Pull request by kezz**
Similar to the `Translatable` interface, this PR adds a `Keybind` interface for implementations with enums/whatever that represent keybinds.

### CODE REVIEW

File: api/src/main/java/net/kyori/adventure/text/KeybindComponent.java [hunk 1]
```diff
@@ -57,6 +58,33 @@
...
+  default @NotNull KeybindComponent keybind(final @NotNull Keybind keybind) {
+    return this.keybind(Objects.requireNonNull(keybind, "keybind").keybind());
+  }
+
+  /**
+   * An object with a keybind.
+   *
+   * @since 4.9.0
+   */
+  interface Keybind {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/KeybindComponent.java:78]: `KeybindSource`? Not sure about the naming.

Comment (kezz) [reply api/src/main/java/net/kyori/adventure/text/KeybindComponent.java:78]: Yeah keybind source probably is better, the thing doesn't *have* to be a keybind after all.

File: api/src/main/java/net/kyori/adventure/text/Component.java [hunk 2]
```diff
@@ -368,6 +368,18 @@
...
 
+  /**
+   * Creates a keybind component with a keybind.
+   *
+   * @param keybind the keybind
+   * @return the keybind component
+   * @since 4.9.0
+   */
+  @Contract(value = "_ -> new", pure = true)
+  static @NotNull KeybindComponent keybind(final @NotNull KeybindComponent.KeybindLike keybind) {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/Component.java:379]:
```suggestion
  static @NotNull KeybindComponent keybind(final KeybindComponent.@NotNull KeybindLike keybind) {
```

File: api/src/main/java/net/kyori/adventure/text/Component.java [hunk 3]
```diff
@@ -381,6 +393,19 @@
...
+  /**
+   * Creates a keybind component with a keybind and styling.
+   *
+   * @param keybind the keybind
+   * @param style the style
+   * @return the keybind component
+   * @since 4.9.0
+   */
+  @Contract(value = "_, _ -> new", pure = true)
+  static @NotNull KeybindComponent keybind(final @NotNull KeybindComponent.KeybindLike keybind, final @NotNull Style style) {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/Component.java:405]:
```suggestion
  static @NotNull KeybindComponent keybind(final KeybindComponent.@NotNull KeybindLike keybind, final @NotNull Style style) {
```

File: api/src/main/java/net/kyori/adventure/text/Component.java [hunk 4]
```diff
@@ -394,6 +419,19 @@
...
+  /**
+   * Creates a keybind component with a keybind, and optional color.
+   *
+   * @param keybind the keybind
+   * @param color the color
+   * @return the keybind component
+   * @since 4.9.0
+   */
+  @Contract(value = "_, _ -> new", pure = true)
+  static @NotNull KeybindComponent keybind(final @NotNull KeybindComponent.KeybindLike keybind, final @Nullable TextColor color) {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/Component.java:431]:
```suggestion
  static @NotNull KeybindComponent keybind(final KeybindComponent.@NotNull KeybindLike keybind, final @Nullable TextColor color) {
```

File: api/src/main/java/net/kyori/adventure/text/Component.java [hunk 5]
```diff
@@ -408,6 +446,20 @@
...
+   * Creates a keybind component with a keybind, and optional color and decorations.
+   *
+   * @param keybind the keybind
+   * @param color the color
+   * @param decorations the decorations
+   * @return the keybind component
+   * @since 4.9.0
+   */
+  @Contract(value = "_, _, _ -> new", pure = true)
+  static @NotNull KeybindComponent keybind(final @NotNull KeybindComponent.KeybindLike keybind, final @Nullable TextColor color, final TextDecoration@NotNull... decorations) {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/Component.java:459]:
```suggestion
  static @NotNull KeybindComponent keybind(final KeybindComponent.@NotNull KeybindLike keybind, final @Nullable TextColor color, final TextDecoration@NotNull... decorations) {
```

File: api/src/main/java/net/kyori/adventure/text/Component.java [hunk 6]
```diff
@@ -422,6 +474,20 @@
...
+   * Creates a keybind component with a keybind, and optional color and decorations.
+   *
+   * @param keybind the keybind
+   * @param color the color
+   * @param decorations the decorations
+   * @return the keybind component
+   * @since 4.9.0
+   */
+  @Contract(value = "_, _, _ -> new", pure = true)
+  static @NotNull KeybindComponent keybind(final @NotNull KeybindComponent.KeybindLike keybind, final @Nullable TextColor color, final @NotNull Set<TextDecoration> decorations) {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/Component.java:487]:
```suggestion
  static @NotNull KeybindComponent keybind(final KeybindComponent.@NotNull KeybindLike keybind, final @Nullable TextColor color, final @NotNull Set<TextDecoration> decorations) {
```

File: api/src/main/java/net/kyori/adventure/text/KeybindComponent.java [hunk 7]
```diff
@@ -57,6 +58,33 @@
...
 
+  /**
+   * Sets the keybind.
+   *
+   * @param keybind the keybind
+   * @return a copy of this component
+   * @since 4.9.0
+   */
+  @Contract(pure = true)
+  default @NotNull KeybindComponent keybind(final @NotNull KeybindComponent.KeybindLike keybind) {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/KeybindComponent.java:69]:
```suggestion
  default @NotNull KeybindComponent keybind(final @NotNull KeybindLike keybind) {
```

File: api/src/main/java/net/kyori/adventure/text/KeybindComponent.java [hunk 8]
```diff
@@ -72,5 +100,17 @@
...
+
+    /**
+     * Sets the keybind.
+     *
+     * @param keybind the keybind
+     * @return this builder
+     * @since 4.9.0
+     */
+    @Contract(pure = true)
+    default @NotNull Builder keybind(final @NotNull KeybindComponent.KeybindLike keybind) {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/KeybindComponent.java:112]:
```suggestion
    default @NotNull Builder keybind(final @NotNull KeybindLike keybind) {
```

**Review by kashike — CHANGES_REQUESTED** [review]

**Review by zml2008 — APPROVED** [review]

**Review by kashike — APPROVED** [review]

**Pull request merged**